escrow/lib.rs

```diff
@@ -0,0 +1,254 @@
+//! # Escrow
+//!
+//! This implements an Escrow contract.
+//!
+//! ## Warning
+//!
+//! This contract is an *example*. It is neither audited nor endorsed for production use.
+//! Do **not** rely on it to keep anything of value secure.
+//!
+//! ## Overview
+//!
+//! Escrow is the third party which holds the asset(asset can be money, bond, stocks)
+//! on the presence of two parties. Escrow will release the fund when certain conditions are met.
+//!
+//! In this contract, there are two parties, a buyer, and a seller. The buyer wants
+//! to buy some goods from the seller, and use this smart contract instance as trusted entity for
+//! depositing funds. The contract is instantiated by the buyer.
+//!
+//! This contract can end up in two ways:
+//! * either delivery of the goods is confirmed by the buyer -> a deposit is transferred to the seller, or
+//! * delivery is canceller by the seller -> a deposit is refunded to the buyer.
+//! In each way contract terminates itself and any remaining funds are transferred to contract
+//! constructor account (the buyer)
+//!
+//! Implementation note: due to how things are implemented in Substrate, a storage deposit for this
+//! contract is returned to message caller. This means that the deposit made by contract instantiation
+//! account, which is the buyer, can be acquired by the seller by calling `refund()`. This value
+//! can be non-negligible on some chains.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+pub mod escrow {
+    use scale::{Decode, Encode};
+
+    #[ink(storage)]
+    pub struct Escrow {
+        /// Buyer's account
+        buyer: AccountId,
+
+        /// Sellers's account
+        seller: AccountId,
+
+        /// Deposit value
+        deposit: Balance,
+    }
+
+    /// Error scenarios in escrow contract
+    #[derive(Debug, PartialEq, Eq, Encode, Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        /// Someone else than buyer tries to confirm
+        ConfirmNotAsBuyer,
+
+        /// Someone else than seller tries to refund
+        RefundNotAsSeller,
+
+        /// requested transfer failed, this can be the case if the contract does not
+        /// have sufficient free funds or if the transfer would have brought the
+        /// contract's balance below minimum balance
+        TransferFailed,
+    }
+
+    /// An event emitted when token transfer occurs
+    #[ink(event)]
+    pub struct Transfer {
+        #[ink(topic)]
+        to: AccountId,
+        value: Balance,
+    }
+
+    pub type Result<T> = core::result::Result<T, Error>;
+
+    impl Escrow {
+        /// Instantiates new escrow contract with buyer as contract author
+        #[ink(constructor, payable)]
```

Comment:
no need to do `payable`, all constructors are implicitly payable

Reply:
Well, maybe not, maybe it's just the outdated documentation ;)

Reply:
It needs to be payable, constructors (just like regular messages) are no longer payable by default